bridge-fungible-token/src/utils.sw

```diff
@@ -19,14 +28,56 @@ const GTF_INPUT_MESSAGE_DATA = 0x11E;
 const GTF_INPUT_MESSAGE_SENDER = 0x115;
 const GTF_INPUT_MESSAGE_RECIPIENT = 0x116;
 
+fn bn_mul(bn: U256, factor: u64) -> (b256, u64) {
+    disable_panic_on_overflow();
+    let product_and_overflow = asm(bn: __addr_of(bn), factor: factor, carry, new_carry, value, current_product, product_with_carry, prod_and_ovf) {
```

Comment:
Why use up two registers (`current_product` and `product_with_carry`) when it was using just one before (the old `v` register)?